src/cursor/cur_table.c

```diff
@@ -837,18 +837,28 @@ __curtable_largest_key(WT_CURSOR *cursor)
 static int
 __curtable_bound(WT_CURSOR *cursor, const char *config)
 {
-    WT_CURSOR *primary;
+    WT_CURSOR **cp, *primary;
+    WT_CURSOR_BOUNDS_STATE saved_bounds;
     WT_CURSOR_TABLE *ctable;
     WT_DECL_RET;
     WT_SESSION_IMPL *session;
+    u_int i;
 
     ctable = (WT_CURSOR_TABLE *)cursor;
+    primary = *ctable->cg_cursors;
     JOINABLE_CURSOR_API_CALL(cursor, session, bound, NULL);
 
-    /* Grab the primary cursor and call bound function. */
-    primary = *ctable->cg_cursors;
-    WT_ERR(primary->bound(primary, config));
+    __wt_cursor_bounds_save(primary, &saved_bounds);
```

Comment:
Add a comment above this: "Save the current state of the bounds in case we fail to apply the new state."

Reply:
[23a8382](https://github.com/wiredtiger/wiredtiger/pull/8265/commits/23a8382b74954e1cbb8d5a7067b5eeb249828ebe)